src/test/java/org/apache/drill_web_test_framework/web_ui/pages/QueryResultsPage.java

```diff
@@ -145,6 +145,19 @@ public String getQueryProfile() {
   public QueryResultsPage showResultRows(String rowsCount) {
     rowsPerPage.click();
     rowsPerPage.findElement(By.xpath("//*[text() = '" + rowsCount + "']")).click();
+    if(!rowsCount.equals("ALL")) {
+      waitForRowsCount((Integer.parseInt(rowsCount) + 1));
+    } else {
+      String allRowsCount = pageRowsInfo.getText()
+          .substring(pageRowsInfo.getText().indexOf("of "))
+          .replaceAll("[a-z ]", "");
```

Comment:
Maybe it's better to use regex fully?

```suggestion
    Matcher matcher = Pattern.compile("([\\d]+)\\w*entries")
        .matcher(pageRowsInfo.getText());
    if (matcher.find()) {
      allRowsCount = matcher.group(1);
    }
```

Reply:
Done